### ISSUE TITLE
Union type doesn't work across modules

### ISSUE THREAD

**Issue by hayeah** [facebook/flow#323]
It's a cross module issue, might be related to 

https://github.com/facebook/flow/issues/283

https://github.com/facebook/flow/issues/286

Here's the minimal case I was able to produce, it allows `foostr` to store either a Foo instance or a string:

```
class Foo {};
var foo = new Foo();
var foostr: Foo | string = foo;
```

It produces no type checking error, as expected. But if we import Foo from another module, we get a type checking error:

```
var Foo = require("./foo");
// No type error if Foo is defined in the same file.
// class Foo {};
var foo = new Foo();
// bug.js:9:19,24: string
// This type is incompatible with
//   foo.js:5:7,9: Foo
var foostr: Foo | string = foo;
```

Where foo.js is:

```
/* @flow */

module.exports = Foo;

class Foo {

}
```

**Comment by hayeah** [comment]
This issue persists as of 9cacbd9

**Comment by avikchaudhuri** [comment]
This should be fixed in the next release, will close when it lands.

**Comment by jeffmo** [comment]
@hayeah : Mind confirming this was fixed in 0.7.0?

**Comment by mroch** [comment]
Reopen if it's still an issue, thanks for the report
